bin/mount-image-callback

```diff
@@ -123,9 +123,10 @@ cleanup() {
 		do_umounts "${UMOUNTS[@]}"
 	fi
 	disconnect_qemu
-	[ -z "${TEMP_D}" -o ! -d "${TEMP_D}" ] ||
-		rm --one-file-system -Rf "${TEMP_D}" ||
-		error "removal of temp dir failed!"
+	if [ -z "${TEMP_D}" -o ! -d "${TEMP_D}" ] &&
+		! grep -q "${TEMP_D}" /proc/mounts; then
```

Comment:
This will have false positives for two reasons:
- it will match mountpoints that start with the same path. /tmp/foo will match /tmp/foobar while /tmp/foobar is not under /tmp/foo. 
- presence of the string in fields other than mountpoint (device or options)

Reply:
Thanks Scott for reviewing, that's the bit when I was most hoping to get feedback. Do you think that's actually a concern, given that `TEMP_D` has a random chunk?
```
TEMP_D=$(mktemp -d "${TMPDIR:-/tmp}/${0##*/}.XXXXXX")
```
What if we make the random part longer, just to be safe?

Reply:
There is also `findmnt --submounts --mountpoint` we could use. I'll check tomorrow what we can do to improve on the current approach.

Reply:
I think `grep "^[^ ]\+ ${TEMP_D}[/ ]" /proc/mounts` will do. I can see only two ways it could fails:
1. Field separator not a single space. The format of `/proc/mounts` is the one described in fstab(5), which says "Fields on each line are separated by tabs or spaces", so that's not forbidden, but I don't think the kernel is ever going to do that.
2. First field has spaces. I wonder if there's anything forbidding this, and I wonder how other tools parsing the file would behave.

Reply:
I created a block device with a whitespace (`mknod "/dev/bad bad" b 8 17`), mounted it and `/proc/mounts` has:
```
/dev/bad\040bad /mnt vfat ...
```
so we're safe from point 2.

Reply:
I added the check you suggested, but also left the `grep /proc/mounts` check, which should be solid and I think worth keeping as we're `rm -Rf`ing stuff.